packages/muelu/src/Misc/MueLu_FilteredAFactory_def.hpp

```diff
@@ -573,7 +576,7 @@ namespace MueLu {
  	if (row >= (LO)numRows) continue;
 	A.getLocalRowView(row, indsA, valsA);
 	for(LO k=0; k<(LO)indsA.size(); k++) {
-	  if(indsA[k] < (LO)numRows) {
+	  if ( (indsA[k] < (LO)numRows) && (valsA[k] != 0.0) ) {
```

Comment:
Instead of checking for actual zeros, should this have a threshold?

Reply:
The hard zero is ok ... but I should do the scalar traits stuff ... that I forgot about. The zero tests whether this is a really a nonzero or not. It is actually somewhat arbitrary (like it changes the math ... but it is not really sure what is better numerically) and does not cause the code to bomb either way. On the other hand, I should be using the scalar traits swill ... and zero() ... and stuff like that. I'll make that change .. hopefully in a few minutes.